Code/Framework/AzFramework/AzFramework/Spawnable/Script/SpawnableScriptMediator.cpp

```diff
@@ -162,18 +162,17 @@ namespace AzFramework::Scripts
             DespawnResult despawnResult;
             despawnResult.m_spawnTicket = spawnTicket;
             m_resultCommands.push_back(despawnResult);
+
+            if (!AZ::TickBus::Handler::BusIsConnected())
```

Comment:
Suggestion: maybe move this logic into a `QueueProcessResults` method to make what this is doing a bit more clear.